src/rdf4cpp/datatypes/registry/util/DateTimeUtils.hpp

```diff
@@ -277,6 +288,51 @@ inline std::chrono::year_month_day normalize(std::chrono::year_month_day i) {
     return static_cast<std::chrono::year_month_day>(static_cast<std::chrono::sys_days>(i + std::chrono::months{0}));
 }
 
+template<std::integral I, I base = 10>
+consteval I number_of_digits(I num) {
+    if (num < 0) {
+        return 1 + number_of_digits(-num);
+    } else if (num < base) {
+        return 1;
+    } else {
+        return 1 + number_of_digits(num / base);
+    }
+}
+static_assert(number_of_digits(0)==1);
+static_assert(number_of_digits(9)==1);
+static_assert(number_of_digits(-1)==2);
+static_assert(number_of_digits(10)==2);
+static_assert(number_of_digits(std::numeric_limits<uint64_t>::max())==std::numeric_limits<uint64_t>::digits10+1);
+struct chrono_max_canonical_string_chars {
+    //std::chrono::year is in [-32767, 32767]
+    static constexpr size_t year = 6;
+    static_assert(std::chrono::year::min() == std::chrono::year(-32767));  //NOLINT
+    static_assert(std::chrono::year::max() == std::chrono::year(32767));   //NOLINT
```

Comment:
why is the year digits not just the digits in the max value + 1? (or max of number of digits in min and max)